newrelic/structures_newrelic_synthetics_script_api_monitor.go

```diff
@@ -22,15 +22,25 @@ func buildSyntheticsScriptAPIMonitorInput(d *schema.ResourceData) synthetics.Syn
 	if attr, ok := d.GetOk("locations_public"); ok {
 		input.Locations.Public = expandSyntheticsPublicLocations(attr.(*schema.Set).List())
 	}
-	if v, ok := d.GetOk("script_language"); ok {
-		input.Runtime.ScriptLanguage = v.(string)
-	}
-	if v, ok := d.GetOk("runtime_type"); ok {
-		input.Runtime.RuntimeType = v.(string)
-	}
-	if v, ok := d.GetOk("runtime_type_version"); ok {
-		input.Runtime.RuntimeTypeVersion = synthetics.SemVer(v.(string))
+
+	_, scriptLangOk := d.GetOk("script_language")
+	_, runtimeTypeOk := d.GetOk("runtime_type")
+	_, runtimeVersionOk := d.GetOk("runtime_type_version")
+
+	if scriptLangOk || runtimeTypeOk || runtimeVersionOk {
+		input.Runtime = &synthetics.SyntheticsRuntimeInput{}
+
```

Comment:
The API is expecting to send all the 3 values so may be changing to `and` would be better.

Reply:
Due to `runtime` being optional at the top level with `runtime.scriptLanguage` being optional and `runtime.runtimeType` and `runtime.runtimeTypVersion` being required, it's a little tricky. First we check to make sure at least one of the values is set, and if it is set, then we can set the `runtime` object. I think this might fix that other weird synthetics issues we were running into a little while back. I'll keep working on this though and add appropriate tests.